### DISCUSSION TITLE
JSON list -- retrieve objects values

### DISCUSSION THREAD

**Discussion by mohammedalasmar** [nlohmann/json#2705] [Q&A] [answered]
How can I get the values from a received JSON format when having an array? 

Example:
 
```
std::string myJsonStr = myJson.dump();
std::cout << myJsonStr << std::endl;

{\"users\":[{\"stringCode\":\"123456\",\"name\":\"joe\",\"time\":1617803265.432}]}
```

In this example, how can I get myJson["name"] (the result is 'joe') and the other fields? 
I tried something like:
`std::string userName = myJson["users"]["name"];`
but this seems to be wrong!

**Comment by nlohmann** [comment, marked as answer]
First off, you can write

```cpp
std::cout << myJson << std::endl;
```

without calling `dump()`.

Then, you can access array elements with an integer index. So it's

```cpp
std::cout << myJson["users"][0]["name"] << std::endl;
```
